packages/gamut-styles/src/system/props/layout.ts

```diff
@@ -0,0 +1,38 @@
+import { DisplayTypes } from '../../variables/display';
+import { parseSize } from '../transforms';
+
+export const layout = {
+  display: {
+    propName: 'display',
+    scale: [] as DisplayTypes[],
+  },
```

Comment:
Capturing Zoom convo into here, this is roughly/vaguely what I was thinking about for creating these...

```suggestion
  display: createScaleProp<DisplayTypes>('display'),
```

More complex ones might use a {} object like:

```ts
createScaleProp<number | string>({
  propName: ['width', 'minWidth', /* ... */],
  computeValue: parseSize
});
```

...where `computeValue` is known to take in a `value: (number | string)` somehow _(is that a doable thing to type in?)_.

What do you think? :smile:

Reply:
Yeah, I've been trying to get something like this to work, but I've been struggling.

I think my grasp on the rules of how TS resolves `unknown` to more defined types is a bit vague, I've got it to work in some cases but others have been more tricky.